app/controllers/carts_controller.rb

```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
 class CartsController < ApplicationController
-
   def show
-    @carts = Product.joins(:cart)
+    @cart_id = session[:cart_id]
+    @products = Product.joins(@cart_id).where.not('cart_id' => nil )
```

Comment:
It would easier to find Cart and then query related products. And you do not need instance variable for `cart` - it do not store any needed data for view except products.
```
cart = Cart.find(session[:cart_id])
@products = cart.products
```

Reply:
> It would easier to find Cart and then query related products. And you do not need instance variable for `cart` - it do not store any needed data for view except products.
> 
> ```
> cart = Cart.find(session[:cart_id])
> @products = cart.products
> ```

I tried it, but it doesn't work for me

Reply:
What I have
![Снимок экрана от 2019-06-18 18-45-15](https://user-images.githubusercontent.com/45793581/59699038-883b9080-91f9-11e9-8efc-01bc463062e3.png)